rcljava_common/src/main/java/org/ros2/rcljava/interfaces/GoalRequestDefinition.java

```diff
@@ -0,0 +1,23 @@
+/* Copyright 2020 Open Source Robotics Foundation, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ros2.rcljava.interfaces;
+
+import java.util.List;
+
+public interface GoalRequestDefinition extends MessageDefinition {
```

Comment:
If instead this would be:

```java
public interface GoalRequestDefinition<T extends ActionDefinition> extends MessageDefinition {
```

I think you could make interfaces like [here](https://github.com/osrf/ros2_java/blob/0f441c390c8b27524b87ed875192ecab9ce0df95/rcljava/src/main/java/org/ros2/rcljava/action/ActionServerImpl.java#L228) really type safe (i.e. checked at compile time that you're passing the `GoalRequestDefinition` from the correct action).

I still need to grok the [other PR](https://github.com/osrf/ros2_java/pull/53/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=.hpp&file-filters%5B%5D=.java&file-filters%5B%5D=.txt&file-filters%5B%5D=.xml), so I may be missing something.

Reply:
Makes sense. Done 4924965